### PR TITLE
New Integration - Extrahop

### PR DESCRIPTION
**Pull request by amshamah419**
## Status
Done

## Related Issues
fixes: demisto/etc#15059

## Description
This is the new integration for ExtraHop. Provides the following:
- Records query
- View Whitelist
- Modify Whitelist
- Create Alerts
- Query Alerts
- Fetch Alerts
- View devices on network

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation https://github.com/demisto/etc/issues/16407
- [x] Code Review

### CODE REVIEW

File: Integrations/ExtraHop/ExtraHop.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    res_raw = http_request('GET', 'alerts')
+    res = res_raw.json()
+    return res
+
+
+def paginate(can_paginate, cursor):
+    while can_paginate is True:
+        body = {
+            "cursor": cursor,
+            "context_ttl": 400000
```
Comment (yuvalbenshalom) [inline Integrations/ExtraHop/ExtraHop.py]: why did you choose 400000?

File: Integrations/ExtraHop/ExtraHop.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+
+def whitelist_retrieve():
+    res_raw = http_request('GET', 'whitelist/devices')
+    res = res_raw.json()
+    return res
+
+
+def add_alert(apply_all, disabled, name, notify_snmp, refire_interval, severity, alert_type, object_type,
+              protocols, field_name, stat_name, units, interval_length, operand, operator, field_name2, field_op,
```
Comment (yuvalbenshalom) [inline Integrations/ExtraHop/ExtraHop.py]: `field_name2` and `param2` are a real thing? :suspect:

File: Integrations/ExtraHop/ExtraHop.py (hunk 1 above)
Comment (amshamah419) [reply Integrations/ExtraHop/ExtraHop.py]: No reason in particular, we can move this to be user configurable if you'd like.

File: Integrations/ExtraHop/ExtraHop.py (hunk 2 above)
Comment (amshamah419) [reply Integrations/ExtraHop/ExtraHop.py]: Sadly yes haha

File: Integrations/ExtraHop/ExtraHop.py [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+            headers=HEADERS,
+            params=payload
+        )
+    except requests.exceptions.RequestException:  # This is the correct syntax
+        return_error('Failed to connect to - {} - Please check the URL'.format(BASE_URL))
+    # Handle error responses gracefully
+    if res.status_code == 204:
+        return demisto.results('Successful Modification')
+    elif res.status_code not in {200, 204, 201}:
+        return_error('Error in API call to ExtraHop {} - {}'.format(res.status_code, res.reason))
```
Comment (yuvalbenshalom) [inline Integrations/ExtraHop/ExtraHop.py]: please switch the error to state the code and reason explicitly, and also go over the other format strings and use named format strings.(most importantly, the ones that have are than one variable. 
example:
`Error in API call. code:{code}; reason: {reason}'.format(code=res.status_code, reason=res.reason)`

**Pull request merged**